src/drf_pydantic/parse.py

```diff
@@ -279,18 +280,22 @@ def _convert_type(  # noqa: PLR0911
                     )
                 elif len(regex_patterns) == 1:
                     return serializers.RegexField(regex=item.pattern, **kwargs)
-            else:
-                for key in [type_, type_.__base__]:
-                    try:
-                        return FIELD_MAP[key](**kwargs)
-                    except KeyError:
-                        continue
+            # Enum
+            elif issubclass(type_, enum.Enum):
+                return serializers.ChoiceField(
+                    choices=[item.value for item in type_], **kwargs
+                )
 
-        # Enum
-        if issubclass(type_, enum.Enum):
-            return serializers.ChoiceField(
-                choices=[item.value for item in type_], **kwargs
-            )
+        # Explicitly defined field
+        class_candidates = [type_]
+        if hasattr(type_, "__mro__"):
+            class_candidates.extend(type_.__mro__)
+
+        for key in class_candidates:
+            try:
+                return FIELD_MAP[key](**kwargs)
+            except KeyError:
+                continue
```

Comment:
I like this. Can you please add some unit tests for this? E.g., a custom class inheriting from a supported type.